WebDriverAgentLib/Categories/XCUIDevice+FBHelpers.m

```diff
@@ -121,9 +125,28 @@ - (NSData *)fb_screenshotWithError:(NSError*__autoreleasing*)error
 #endif
 }
 
-- (NSData *)fb_rawScreenshotWithQuality:(NSUInteger)quality rect:(CGRect)rect error:(NSError*__autoreleasing*)error
+- (NSData *)fb_rawScreenshotWithQuality:(NSUInteger)quality rect:(CGRect)rect error:(NSError*__autoreleasing*) error
 {
-  return [XCUIScreen.mainScreen screenshotDataForQuality:quality rect:rect error:error];
+  id<XCTestManager_ManagerInterface> proxy = [FBXCTestDaemonsProxy testRunnerProxy];
+  BOOL fasterAPISupported = [(NSObject *) proxy respondsToSelector:@selector(_XCT_requestScreenshotOfScreenWithID:withRect:uti:compressionQuality:withReply:)];
+  __block NSData *screenshotData = nil;
+  dispatch_semaphore_t sem = dispatch_semaphore_create(0);
+  void (^completion)(NSData *, NSError *) = ^(NSData *data, NSError *screenshotError) {
+    screenshotData = data;
+    *error = screenshotError;
+    dispatch_semaphore_signal(sem);
+  };
+  if (fasterAPISupported) {
+    [proxy _XCT_requestScreenshotOfScreenWithID:[[XCUIScreen mainScreen] displayID]
+                                       withRect:CGRectNull
```

Comment:
why rect argument is ignored?

Reply:
I can remove the argument actually. We always pass CGRectNull as well

Reply:
if this is the only value we use there then yes, it's better to remove the argument